### PR TITLE
add getting-started-openshift-s2i (fixes #1306)

### LINKED ISSUE #1306: Add documentation for building with S2I on OpenShift
The current docs on the website use Dockerfile for building the app image:
https://quarkus.io/guides/kubernetes-guide

There should also be docs on how to use S2I and OpenJDK builder image for building non-native and possibly also native packaging on OpenShift.

### PR DESCRIPTION
**Pull request by vorburger**
https://github.com/quarkusio/quarkus/issues/1306

see also https://github.com/quarkusio/quarkus/issues/1305

requires https://github.com/quarkusio/quarkus-quickstarts/pull/81

and then to actually publish the link to this new guide in the menu
it needs https://github.com/quarkusio/quarkusio.github.io/pull/126/

### CODE REVIEW

File: docs/src/main/asciidoc/openshift-s2i-guide.adoc
Comment (vorburger) [inline docs/src/main/asciidoc/openshift-s2i-guide.adoc]: This (and the next paragraph) should be removed, given that I wrote "You do not need to locally clone the Git repository, as it will be directly built since OpenShift." below, agreed @tqvarnst @siamaksade?

File: docs/src/main/asciidoc/openshift-s2i-guide.adoc
Comment (vorburger) [inline docs/src/main/asciidoc/openshift-s2i-guide.adoc]: The `context-dir` here will have to be changed, following the decision in https://github.com/quarkusio/quarkus-quickstarts/pull/81 ...

**Comment by vorburger** [comment]
@cescoffier merge this after https://github.com/quarkusio/quarkus-quickstarts/pull/81 is in? -- Then close https://github.com/quarkusio/quarkus/issues/1305.

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
Waiting to get https://github.com/quarkusio/quarkus-quickstarts/pull/81 figured out before merging.

**Pull request merged**
